Source/Graphic/HCPath.c

```diff
@@ -230,21 +205,362 @@ HCPathElement HCPathElementAt(HCPathRef self, HCInteger elementIndex) {
+        HCBoolean closed = HCPathContourContainingElementIsClosed(self, elementIndex, &startIndex, &endIndex);
+        elementIndex = endIndex;
+        
+        // If closed, create the contour path with the elements from start to end
+        if (closed) {
+            HCPathElement* elements = (HCPathElement*)HCDataBytes(self->elementData);
+            HCPathRef contourPath = HCPathCreateWithElements(elements + startIndex, endIndex - startIndex);
+            if (!HCPathIsEmpty(contourPath)) {
+                HCListAddObjectReleased(closedContourPaths, contourPath);
+            }
+        }
+    }
+    return closedContourPaths;
+}
+
+HCPathRef HCPathOpenContoursAsPathRetained(HCPathRef self) {
+    HCListRef openContours = HCPathOpenContourPathsRetained(self);
+    HCPathRef openPath = HCPathCreateWithSubpaths(openContours);
+    HCRelease(openContours);
+    return openPath;
 }
 
-HCInteger HCPathElementPolylinePointCount(HCPathRef self, HCInteger elementIndex) {
-    HCDataRef polylineData = HCPathElementPolylineData(self, elementIndex);
+HCPathRef HCPathClosedContoursAsPathRetained(HCPathRef self) {
+    HCListRef closedContours = HCPathClosedContourPathsRetained(self);
+    HCPathRef closedPath = HCPathCreateWithSubpaths(closedContours);
+    HCRelease(closedContours);
+    return closedPath;
+}
+
+//----------------------------------------------------------------------------------------------------------------------------------
+// MARK: - Polylines
+//----------------------------------------------------------------------------------------------------------------------------------
+HCInteger HCPathPolylineCount(HCPathRef self) {
+    return HCPathElementCount(self);
+}
+
+HCInteger HCPathPolylinePointCount(HCPathRef self, HCInteger polylineIndex) {
+    HCDataRef polylineData = HCListObjectAtIndex(self->polylines, polylineIndex);
     return HCDataSize(polylineData) / sizeof(HCPoint);
 }
 
-HCPoint HCPathElementPolylinePointAt(HCPathRef self, HCInteger elementIndex, HCInteger pointIndex) {
-    HCDataRef polylineData = HCPathElementPolylineData(self, elementIndex);
+HCPoint HCPathPolylinePointAt(HCPathRef self, HCInteger polylineIndex, HCInteger pointIndex) {
+    HCDataRef polylineData = HCListObjectAtIndex(self->polylines, polylineIndex);
     HCPoint* points = (HCPoint*)HCDataBytes(polylineData);
     return points[pointIndex];
 }
 
+const HCPoint* HCPathPolylineAt(HCPathRef self, HCInteger polylineIndex) {
+    return (HCPoint*)HCDataBytes(HCListObjectAtIndex(self->polylines, polylineIndex));
+}
+
+//----------------------------------------------------------------------------------------------------------------------------------
+// MARK: - Polyline / Path Element Correspondence
+//----------------------------------------------------------------------------------------------------------------------------------
+HCInteger HCPathIndexOfPolylineContainingElement(HCPathRef self, HCInteger elementIndex) {
+    (void)self; // unused
```

Comment:
Do we need this?

Reply:
Yes, otherwise compiling with -Wall gives a warning.